### ISSUE TITLE
Does the path `~~exe_dir/` works normally?

### ISSUE THREAD

**Issue by hooke007** [mpv-player/mpv#10009]
### Important Information

Provide following Information:
- mpv version
 https://github.com/shinchiro/mpv-winbuild-cmake/releases/tag/20220323
- Windows Version
 win11 21h2
- Source of the mpv binary
 https://github.com/shinchiro/mpv-winbuild-cmake/releases/tag/20220323
- If known which version of mpv introduced the problem
 N/A
- Possible screenshot or video of visual glitches
 see below

### Reproduction steps

Original issue: https://github.com/shinchiro/mpv-packaging/issues/21
The behaviour shows different with the first time. Now it cannot output at all.

1. install mpv by the bat file
2. write the contents into mpv.conf
```
 log-file = "~~exe_dir/temp/mpv.log"
 screenshot-directory = "~~exe_dir/temp/"
 sharpen=2
 gpu-shader-cache-dir="~~exe_dir/temp/"
```
3. open any video (i.e. a file in desktop) with mpv (windows context menu)
4. try to create the screenshot

### Expected behavior

The logfile/screenshot/caches should be saved where mpv.exe is located.

### Actual behavior

Nothing could be saved.
![Snipaste_2022-03-23_16-17-34](https://user-images.githubusercontent.com/41094733/159747927-95e072bf-1be4-473e-ba18-ce724653eb61.jpg)

### Log file

In this case it could not output the log.

### Sample files

N/A

**Comment by shinji257** [comment]
It works but you have to create the folder you are pointing to first otherwise it errors as shown above.

**Comment by hooke007** [comment]
> create the folder you are pointing to

Doesn't work , No diff here.

**Comment by Dudemanguy** [comment]
Old but what's likely happening here is that `exe_dir` is returning `""` for some reason instead of an actual path for you. Looks like a bug.

**Comment by sharpchen** [comment]
`~~exe_dir/` expands to empty string on portable mode in my case.

**Comment by verygoodlee** [comment]
> `~~exe_dir/` expands to empty string on portable mode in my case.

Yes, in portable mode, only `~~home/` `~~cache/` and `~~desktop/` will be expanded.
https://github.com/mpv-player/mpv/blob/174b638f297626dc247d6436bd9ad0340be03a7c/osdep/path-win.c#L95-L119

@Dudemanguy Is it a bug or an intentional design?
I can only confirm that `exe_dir` is designed to replace `old_home`. 26fc70dbfdb11844497f34e2d96ecd6a011cd6bc
